x/amm/spec/06_swap_txs_reordering.md

```diff
@@ -0,0 +1,35 @@
+# Swap transactions reordering
+
+To provide lowest slippage as possible on the block, module sort the swap requests into a specific order.
+
+This removes possibility of MEV attack on swap transactions.
+
+Instead of executing messages directly, on the endblocker, we store the requests on transient store and execute in specific order to minimize maximum slippage.
```

Comment:
@jelysn 2 concerns from my end.
1. batch execution will give benefit to Elys or traders?
2. sort execution order has to be deterministic in every where.
3. How can we handle txs that are related? Shall we process it in the next block?
let's say we have 10 swap request per block, all of them wants convert elys to usdc. after 9th transaction, our elys pool will run out. Shall we group the only the first 8 txs and execute them and fail to the rest 2? how can we calculate the final slippage?

Reply:
1. Batch execution will give benefit to both traders and to Elys - saving liquidity providers from arbitrage users & giving lower slippage by combining with other requests to reduce slippage.
2. Yes, this is going to use deterministic functions everywhere
3. 9th transaction will fail for minimum out amount restriction that user set. Here, batch execution means store all the requests, and execute at once. Not combine those transactions into one. We were going to combine txs for batch execution, but this is not easy and therefore changed the mechanism.